### PR TITLE
Get rid of heavy dependencies

### PR DESCRIPTION
**Pull request by maksim-tolo**

### CODE REVIEW

**Comment by Arachnid** [comment]
Thanks! Can you please update your PR to the new master?

File: lib/eos/ecc/point.js [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
+import { assert } from './../helper';
 
-var THREE = BigInteger.valueOf(3)
+const ONE = bigInt.one;
+const ZERO = bigInt.zero;
+const THREE = bigInt(3);
 
-function Point (curve, x, y, z) {
-  assert.notStrictEqual(z, undefined, 'Missing Z coordinate')
+function testBit(bigIntVal, index) {
```
Comment (Arachnid) [inline lib/eos/ecc/point.js]: This is horrendously inefficient. Can't you use bitmasks for this?

File: lib/eos/ecc/point.js [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
+  return bigIntVal.toString(2)[index] === '1';
+}
+
+function bufferToBigInt(buffer) {
+  return bigInt(buffer.toString('hex'), 16);
```
Comment (Arachnid) [inline lib/eos/ecc/point.js]: Does `bigInt` not have a constructor that accepts a buffer?

File: lib/eos/ecc/point.js [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
+}
+
+function bigIntToBuffer(bigIntVal, size) {
+  return Buffer.from(bigIntVal.toString(16).padStart(size, '0'), 'hex');
```
Comment (Arachnid) [inline lib/eos/ecc/point.js]: Likewise - does it not have a function that returns a buffer?

File: lib/sha256/sha256.js [hunk 4]
```diff
@@ -8,9 +8,22 @@
  *
  */
 
-var inherits = require('inherits')
-var Hash = require('./hash')
-var Buffer = require('safe-buffer').Buffer
+import { Buffer } from 'safe-buffer';
+import Hash from './hash';
+
+function inherits(ctor, superCtor) {
```
Comment (Arachnid) [inline lib/sha256/sha256.js]: Where is this used?

File: lib/eos/ecc/point.js (hunk 1 above)
Comment (maksim-tolo) [reply lib/eos/ecc/point.js]: Fixed

File: lib/eos/ecc/point.js (hunk 2 above)
Comment (maksim-tolo) [reply lib/eos/ecc/point.js]: Unfortunately `bigInt` doesn't have a built-in constructor that accepts a buffer

File: lib/eos/ecc/point.js (hunk 3 above)
Comment (maksim-tolo) [reply lib/eos/ecc/point.js]: No, it doesn't

File: lib/sha256/sha256.js (hunk 4 above)
Comment (maksim-tolo) [reply lib/sha256/sha256.js]: `inherits(Sha256, Hash)`

File: lib/eos/ecc/point.js (hunk 2 above)
Comment (Arachnid) [reply lib/eos/ecc/point.js]: I think `toArray` and `fromArray` with radix 256 should work here.

Comment (maksim-tolo) [reply lib/eos/ecc/point.js]: Fixed

**Pull request merged**